include/sys/arc_impl.h

```diff
@@ -885,7 +952,7 @@ typedef struct arc_evict_waiter {
 #define	ARCSTAT(stat)	(arc_stats.stat.value.ui64)
 
 #define	ARCSTAT_INCR(stat, val) \
-	atomic_add_64(&arc_stats.stat.value.ui64, (val))
+	wmsum_add(&arc_wmsums.stat, (val))
```

Comment:
Consider using:
```
       wmsum_add(&arc_sums.sum_##stat, (val))
```
and then changing callers to just pass the tail of the field name:
```
        ARCSTAT_INCR(data_size, space);
```

Reply:
For new code I would more or less agree, but now it would require changes to every consumer.  Not sure I want so big patch.